x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_confirmation_modal.test.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import '../../../../__mocks__/kea.mock';
+import { setMockActions, setMockValues } from '../../../../__mocks__';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { shallow } from 'enzyme';
+
+import { LogRetentionConfirmationModal } from './log_retention_confirmation_modal';
+import { ELogRetentionOptions } from './types';
+import { GenericConfirmationModal } from './generic_confirmation_modal';
+
+describe('<LogRetentionConfirmationModal />', () => {
+  beforeAll(() => {
+    // LogRetentionConfirmationModal contains EuiModals, which utilize React Portals,
+    // so we must mock `createPortal` to get the rendered element directly,
+    // instead of letting it be placed normally elsewhere in DOM (outside of jest's domain)
+    // @ts-ignore
+    ReactDOM.createPortal = jest.fn((element) => {
+      return element;
+    });
```

Comment:
Can we do this at the top of the file instead of overriding ReactDOM.createPortal?

```
jest.mock('ReactDOM', () => ({
  createPortal: jest.fn((element) => element),
})
```

Reply:
Actually I'm very confused. This file shouldn't actually contain EuiModals directly at all especially since we're `shallow()`ing, so the tests _shouldn't_ (in theory) be diving deep enough down to render a portal. Can we lose this entirely?

Reply:
It's just carry-over so I didn't touch it. I will investigate.

Reply:
Ohhhh I see. The tests were previously doing full `render`s. I changed it to `shallow`. So yes, those are no longer necessary.

Reply:
https://github.com/elastic/kibana/pull/83009/commits/99463bb6ab73cc0db72df7b445e5ab669e55380e